go.mod

```diff
@@ -28,9 +28,9 @@ require (
 	github.com/ipld/go-ipld-prime v0.21.0
 	github.com/ipld/go-trustless-utils v0.4.1
 	github.com/ipld/ipld/specs v0.0.0-20231012031213-54d3b21deda4
-	github.com/ipni/go-libipni v0.5.7
-	github.com/ipni/storetheindex v0.8.9
-	github.com/libp2p/go-libp2p v0.32.1
+	github.com/ipni/go-libipni v0.5.4
+	github.com/ipni/storetheindex v0.8.5
+	github.com/libp2p/go-libp2p v0.31.0
```

Comment:
It would be nice to get `github.com/libp2p/go-libp2p v0.31.1` since it has a memory leak fix.